ank/src/main.rs

```diff
@@ -151,7 +153,8 @@ async fn main() {
             None => unreachable!("Unreachable code."),
         },
         cli::Commands::Apply(apply_args) => {
-            if let Err(err) = cmd.apply_manifests(apply_args).await {
+            let handle_apply_args: Box<dyn HandleApplyArgs> = Box::new(apply_args.clone());
+            if let Err(err) = cmd.apply_manifests(apply_args, handle_apply_args).await {
```

Comment:
This part is really strange. I understand why it was needed - to enable partial mock of the arguments, s.t. some part are used as values and the functionality is mocked.

Before writing a useless comment asking for a change that could be impossible, I started trying out if it is possible to separate the data from the function in order to enable the mocking. At the end it worked with some mocking of free utility functions: https://github.com/eclipse-ankaios/ankaios/pull/299/commits/2328eb4625359985adb60df722b466c62aeaa6f0
Sorry for directly messing in the branch, but I wanted to be sure it will work 😉